validator/accounts/v2/wallet/wallet.go

```diff
@@ -100,35 +100,82 @@ func New(cfg *Config) *Wallet {
 	}
 }
 
-// Exists check if a wallet at the specified directory
-// exists and has valid information in it.
-func Exists(walletDir string) error {
+// Exists checks if directory at walletDir exists
+func Exists(walletDir string) (bool, error) {
 	dirExists, err := fileutil.HasDir(walletDir)
 	if err != nil {
-		return errors.Wrap(err, "could not parse wallet directory")
+		return false, errors.Wrap(err, "could not parse wallet directory")
 	}
 	if dirExists {
-		isEmptyWallet, err := isEmptyWallet(walletDir)
-		if err != nil {
-			return errors.Wrap(err, "could not check if wallet has files")
+		return true, nil
+	}
+	return false, nil
+}
+
+// isValidWallet checks if a folder contains a key directory such as `derived`, `remote` or `direct`.
+// Returns true if one of those subdirectories exist, false otherwise.
+func isValidWallet(walletDir string) (bool, error) {
+	expanded, err := fileutil.ExpandPath(walletDir)
+	if err != nil {
+		return false, err
+	}
+	f, err := os.Open(expanded)
+	if err != nil {
+		return false, err
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			log.Debugf("Could not close directory: %s", expanded)
 		}
-		if isEmptyWallet {
-			return ErrNoWalletFound
+	}()
+	names, err := f.Readdirnames(-1)
+	if err == io.EOF {
+		return false, nil
+	}
+
+	for _, n := range names {
```

Comment:
There can only be one wallet in a folder (#7290), so you should verify that `len(names) == 1` and check the name.

Reply:
Since the wallet may have a file called hash (but remote wallet type may or may not (https://discordapp.com/channels/476244492043812875/730447046930071672/760305488029155350) length of that slice for a valid wallet may be 1 or 2.  Anyways I adjusted the code a little to work with that.